library/L2_HAL/sensors/environment/temperature/test/si7060_test.cpp

```diff
@@ -1,13 +1,211 @@
+          .position   = 0,
+          .repeated   = false,
+          .busy       = true,
+          .timeout    = I2c::kI2cTimeout,
+          .status     = Status::kSuccess,
+      }),
+    };
+
+    // Array to store the two transactions.
+    static I2c::Transaction_t * transactions =
+        new I2c::Transaction_t[kExpectedTransactions.size()];
+    // Storing Transaction's data_out in separate array since it's read-only.
+    static uint8_t data_out[kExpectedTransactions.size()][kMaxDataLength];
+    // Stub Transaction to extract each Transaction record and to inject data to
+    // receive buffer when temperature data is expected to be received.
+    When(Method(mock_i2c, Transaction))
+        .AlwaysDo([=](I2c::Transaction_t transaction) -> Status {
+          static uint8_t transaction_id = 0;
+          transactions[transaction_id]  = transaction;
+          for (size_t i = 0; i < transaction.out_length; i++)
+          {
+            data_out[transaction_id][i] = transaction.data_out[i];
+          }
+          // Inject the temperature data for the 3rd and 4th transactions.
+          constexpr uint8_t kTransactionGetMostSignificantByte  = 2;
+          constexpr uint8_t kTransactionGetLeastSignificantByte = 3;
+          switch (transaction_id)
+          {
+            case kTransactionGetMostSignificantByte:
+              transaction.data_in[0] = kExpectedTemperatureData[0];
+              break;
+            case kTransactionGetLeastSignificantByte:
+              transaction.data_in[0] = kExpectedTemperatureData[1];
+              break;
+            default: break;
+          }
+          transaction_id++;
+          return Status::kSuccess;
+        });
+
+    units::temperature::celsius_t temperature;
+    temperature_sensor.GetTemperature(&temperature);
+    // Verify the configuration and data in the four transactions.
+    Verify(Method(mock_i2c, Transaction)).Exactly(kExpectedTransactions.size());
+    for (size_t i = 0; i < kExpectedTransactions.size(); i++)
+    {
+      INFO("Checking transaction: #" << (i + 1));
+      CHECK(transactions[i].operation == kExpectedTransactions[i].operation);
+      CHECK(transactions[i].address == kExpectedTransactions[i].address);
+      CHECK(transactions[i].out_length == kExpectedTransactions[i].out_length);
+      CHECK(transactions[i].in_length == kExpectedTransactions[i].in_length);
+      CHECK(transactions[i].position == kExpectedTransactions[i].position);
+      CHECK(transactions[i].repeated == kExpectedTransactions[i].repeated);
+      CHECK(transactions[i].timeout == kExpectedTransactions[i].timeout);
+      CHECK(transactions[i].status == kExpectedTransactions[i].status);
+      for (size_t j = 0; j < transactions[i].out_length; j++)
+      {
+        CHECK(data_out[i][j] == kExpectedTransactions[i].data_out[j]);
+      }
+    }
+
+    const float kError = fabs(temperature.to<float>() - kExpectedTemperature);
+    CHECK(0.0f <= kError);
+    CHECK(kError < 0.001f);
```

Comment:
Replace this with the usage of the matcher `Approx()`. 

See: https://github.com/catchorg/Catch2/blob/master/docs/matchers.md
Example Test Usage: library/L2_HAL/sensors/battery/test/ltc4150_test.cpp